README.md

```diff
@@ -14,9 +14,21 @@ Online documentation is available at https://mozilla.github.io/mozanalysis/
 pip install mozanalysis
 ```
 
+## Local Installation
+```bash
+# Create and activate a python virtual environment.
+python3 -m venv venv/
+source venv/bin/activate
+pip install -r requirements.txt
+pip install -e .
+```
+
 ## Development
+After installing the dependencies as described above, install the package and additional development dependencies with `pip install -e ".[dev]"`
 Linting and Formatting are done with Ruff.
 
+When adding new dependencies, add them to the `pyproject.toml` `dependencies` list.  Then update the `requirements.txt` file using the script `script/update_deps`.  [`pip-compile` is called](https://pypi.org/project/pip-tools/) in this script, which uses the specified dependencies to create the `requirements.txt` file.
```

Comment:
Ideally all dependencies are kept up to date, but in the event that doesn't happen (like with Jetstream, which we go through and update periodically, but sometimes there can be a dozen or two dependencies that have available updates), should we document a way for someone to update a single dependency?

In Jetstream, that means manually editing `requirements.in` with the new version of the package you want to use. Not sure if you have a better way, but could add that here too.

Reply:
I would say if someone needs to do that they should specify a minimum version for whatever dependency they want to update in the `pyproject.toml` and then update with the standard command.  Putting the minimum version provides some self documentation and I think there is value in having to update the dependencies "all the way" rather than doing a kind of bandaid.  What do you think?

Reply:
Agreed!